Multisig/Data/Services/Utils/HTTPClient.swift

```diff
@@ -4,61 +4,95 @@
 
 import Foundation
 
-public protocol HTTPRequest {
-
+protocol HTTPRequest {
     var httpMethod: String { get }
     var urlPath: String { get }
     var query: String? { get }
     var body: Data? { get }
     var headers: [String: String] { get }
 }
 
-public extension HTTPRequest {
-
+extension HTTPRequest {
     var query: String? { return nil }
     var body: Data? { return nil }
     var headers: [String: String] { return [:] }
-
 }
 
 /// Synchronous http client
-public class HTTPClient {
-    /// Client error
+class HTTPClient {
+    private let baseURL: URL
+    private let logger: Logger?
+
+    private typealias URLDataTaskResult = (data: Data?, response: URLResponse?, error: Swift.Error?)
+
+    /// Creates new client with baseURL and logger
     ///
-    /// - networkRequestFailed: network request failed for some reason. Provided are request, response and data values.
-    public enum Error: LocalizedError {
+    /// - Parameters:
+    ///   - url: base url for creating all request urls
+    ///   - logger: logger for debugging and error purposes
+    init(url: URL, logger: Logger? = nil) {
+        baseURL = url
+        self.logger = logger
+    }
+
+    enum Error: LocalizedError {
         case networkRequestFailed(URLRequest, URLResponse?, Data?)
         case entityNotFound(URLRequest, URLResponse?, Data?)
         case unprocessableEntity(URLRequest, URLResponse?, Data?)
         case unknownError(URLRequest, URLResponse?, Data?)
 
-        public var errorDescription: String? {
+        enum Message: String {
+            case networkRequestFailed = "No internet connection."
```

Comment:
is this really the case that if a request fails, then it is always because of no internet?

Reply:
you are right. No internet connection description is provided by the system.
I will fix this message.